### PR TITLE
feat: full screen previewer

### LINKED ISSUE #4764: Full screen button gallery item
## Desktop
image
- the button is displayed when you hover over the img. 
- on big touch devices its there by default 
- once you click you will be redirected to a blank page with a full screen img and a option to go back
image
- this screen is same for mobile
## Touch devices (mobile)
image
full screen button over the img

## Buttons
image
- feel free to change the icon from the library we are using
- the dark is for dark mode mobile, the light is for light mode mobile. 
- button are in the gallery item handoff figma: https://www.figma.com/file/gHSzqGPeQ9qK4L0qIDcTjY/gallery-item-handoff?node-id=3454%3A4104&t=yzdLQ1QNjc098YaY-0
- 
Thanks!

### PR DESCRIPTION
**Pull request by floyd-li**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

this pr is to add the full screen previewer feature on gallery item, also fix some display issue i commited in #5309

FYI: @prachi00 since the break point is set to 930px so i remove the styles you commit in #5134, it caused some display issue in new break point

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #4764
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EZiu1PjV2j2JHKxY6mHnFwwCRCoV27uHKQSkKXATSh1srJT)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
image
image
image
image

### CODE REVIEW

**Comment by floyd-li** [comment]
@exezbcz please do the ui check :)

File: components/gallery/GalleryItem.vue [hunk 1]
```diff
@@ -96,11 +99,39 @@
...
+              resolveMedia(nftMimeType) == MediaType.AUDIO,
+          }"
+          class="gallery-item-media"
+          :src="nftImage"
+          :animation-src="nftAnimation"
+          :mime-type="nftMimeType"
+          :title="nft?.name || nft?.id"
+          original />
+      </div>
+    </b-modal>
```
Comment (roiLeo) [inline components/gallery/GalleryItem.vue:129]: could you make it as a component?
why do we have a part related with audio media?

**Review by roiLeo — COMMENTED** [review]
Hey, small note

**Comment by exezbcz** [comment]
hey hey, nice 
image
- can you make the buttons look like this please? 
- they are 35px 
- they are in figma - gallery item handoff, icon section https://www.figma.com/file/gHSzqGPeQ9qK4L0qIDcTjY/gallery-item-handoff?node-id=1270%3A5615&t=rutwiIgwZu2HqYYh-1

**Comment by roiLeo** [comment]
Why did you change `NeoIcon` to use svg? I prefer that you use fontawesome `expand` icon since it's the same and then change color according to theme

**Comment by floyd-li** [comment]
> Why did you change `NeoIcon` to use svg? I prefer that you use fontawesome `expand` icon since it's the same and then change color according to theme

reverted using `NeoIcon` @roiLeo

File: components/gallery/GalleryItem.vue (hunk 1 above)
Comment (exezbcz) [reply components/gallery/GalleryItem.vue]: @floyd-li ye, that makes sense, audio only has cover img that does not really need fullscreen button tbh

**Comment by exezbcz** [comment]
+ can you please make the button visible on all touch devices - not only mobile 
image

File: components/gallery/GalleryItem.vue [hunk 2]
```diff
@@ -212,9 +227,34 @@ useNuxt2Meta({
...
+
+.dark-mode .fullscreen-button {
+  color: #fff !important;
+  background: rgba(0, 0, 0, 0.15);
+  border-color: rgba(0, 0, 0, 0.3);
+}
+
+.column:hover .fullscreen-button {
+  display: flex;
+}
```
Comment (roiLeo) [inline components/gallery/GalleryItem.vue:252]: Do you really need to set hover on column?
or just wrap mediItem in a new div

File: components/gallery/GalleryItem.vue [hunk 3]
```diff
@@ -212,9 +227,34 @@ useNuxt2Meta({
...
+  color: #000 !important;
+  background: rgba(255, 255, 255, 0.15);
+  border-color: rgba(255, 255, 255, 0.3);
+}
+
+.dark-mode .fullscreen-button {
+  color: #fff !important;
+  background: rgba(0, 0, 0, 0.15);
+  border-color: rgba(0, 0, 0, 0.3);
+}
```
Comment (roiLeo) [inline components/gallery/GalleryItem.vue:248]: check out `@include ktheme()` for theming color, and if possible try to avoid using `!important`

**Review by roiLeo — COMMENTED** [review]
otherwise wfm ✅

File: components/gallery/GalleryItem.vue [hunk 4]
```diff
@@ -134,6 +142,11 @@ const tabs = {
 const activeTab = ref(tabs.offers)
 const showCongratsMessage = ref(false)
 
+const isFullscreen = ref(false)
+const canPreview = computed(() =>
+  [MediaType.VIDEO, MediaType.IMAGE].includes(resolveMedia(nftMimeType.value))
```
Comment (roiLeo) [inline components/gallery/GalleryItem.vue:147]: Maybe add `MediaType.OBJECT` too

**Comment by exezbcz** [comment]
last thing: 
> can you please make the button visible on all touch devices - not only mobile

this still does not work, it should be shown by default, not after click on the img

otherwise good!

**Comment by floyd-li** [comment]
updated

**Review by roiLeo — APPROVED** [review]
✅ wfm

**Review by roiLeo — CHANGES_REQUESTED** [review]
Mobile navbar changed

**beta**
Screenshot 2023-03-23 at 17-51-32 Mountain KodaDot - NFT Market Explorer


**preview**
Screenshot 2023-03-23 at 17-51-47 Mountain KodaDot - NFT Market Explorer

File: components/gallery/GalleryItem.vue [hunk 5]
```diff
@@ -10,19 +10,27 @@
...
+              'is-flex is-align-items-center is-justify-content-center h-audio':
+                resolveMedia(nftMimeType) == MediaType.AUDIO,
+            }"
+            class="gallery-item-media"
+            :src="nftImage"
+            :animation-src="nftAnimation"
+            :mime-type="nftMimeType"
+            :title="nftMetadata?.name"
+            is-detail
+            :original="isMobile && true" />
```
Comment (Jarsen136) [inline components/gallery/GalleryItem.vue:32]:
```suggestion
            :original="isMobile" />
```

File: components/gallery/GalleryItemPreviewer.vue [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  <b-modal
+    v-model="isFullscreen"
+    :destroy-on-hide="false"
+    :can-cancel="false"
+    full-screen
+    custom-class="gallery-item-modal"
+    custom-content-class="gallery-item-modal-content">
+    <NeoButton class="back-button" @click.native="emit('input', false)">
+      <NeoIcon icon="chevron-left" />
+      {{ $i18n.t('go back') }}
```
Comment (Jarsen136) [inline components/gallery/GalleryItemPreviewer.vue:11]:
```suggestion
      {{ $t('go back') }}
```

**Review by Jarsen136 — COMMENTED** [review]
some small stuffs. Otherwise lgtm

**Comment by floyd-li** [comment]
> Mobile navbar changed

hmm seems it's the result of the break point change, i'll fix it

File: styles/layouts/_main-navigation.scss [hunk 7]
```diff
@@ -131,19 +131,19 @@
   transition: 0.3s ease;
   -webkit-transition: 0.3s ease;
   z-index: 999 !important;
-  min-height: 84px;
+  min-height: $navbar-desktop-min-height;
```
Comment (roiLeo) [inline styles/layouts/_main-navigation.scss:134]: this part of code can be a problem (responsive) but it's not related with this issue, lets keep it for another one

**Review by roiLeo — APPROVED** [review]
✅ wfm

File: styles/layouts/_main-navigation.scss (hunk 7 above)
Comment (vikiival) [reply styles/layouts/_main-navigation.scss:134]: If it breaks ui let's resolve it now please

**Review by vikiival — CHANGES_REQUESTED** [review]

File: styles/layouts/_main-navigation.scss (hunk 7 above)
Comment (roiLeo) [reply styles/layouts/_main-navigation.scss:134]: this was introduce before this PR (maybe https://github.com/kodadot/nft-gallery/commit/3433b00ec2b1f1233afd0497998ef5d6ee3071de & changed in https://github.com/kodadot/nft-gallery/pull/5093)

**Comment by vikiival** [comment]
pay 35 usd
